### ISSUE TITLE
Not working with electron 7.x

### ISSUE THREAD

**Issue by vbourgeois** [arantes555/electron-fetch#17]
Hello,

When using the library with the `useElectronNet=true` option, any request working with Electron 6.x is now throwing an error with electron 7.x, here it is : 

```
FetchError: request to https://xxx/graphql failed, reason: net::ERR_INVALID_ARGUMENT
    at ClientRequest.<anonymous> (/home/vincent/lls/lls-desktop/node_modules/electron-fetch/lib/index.js:1415:13)
    at ClientRequest.emit (events.js:203:13)
    at URLRequest../lib/browser/api/net.js.URLRequest._emitRequestEvent (electron/js2c/browser_init.js:2284:24)
```

Any hint on this?

Thanks,
Vincent

**Comment by arantes555** [comment]
Hello @vbourgeois !

Yeah, I noticed this yesterday, when trying to run the tests with electron7. Everything seems to fail, except GET requests... It looks like the update basically broke the net module. Or maybe there is a breaking change somewhere that did not make it in the changelog? The electron team has done this sort of things before...

In any case, I'm sorry but I do not have a solution for you at the moment apart from staying on the 6.x branch (which has a few bugs in `net` too, but much less). I will investigate this more in the next few days and keep you updated.

**Comment by vbourgeois** [comment]
Hello,

You're right it seems to be an issue with the net module from electron, I have the same problem with another module `got` (using useElectronNet option).
Didn't see anything in the changelog about breaking changes in the net module for electron 7...

Thanks for your help.

**Comment by arantes555** [comment]
@vbourgeois after a bit more investigation, it seems the issue lies in that electron@7 prevents you for setting `content-length` in a request. It is something `electron-fetch` does, to stick to the fetch specification as closely as possible.

I opened an issue with electron here : https://github.com/electron/electron/issues/21091 

I'll wait a few days to see how they respond to the report. If it does not move on their side (given their track record with bugs in `net`, I'm not very hopeful...), I think I'll be able to implement a work-around in `electron-fetch`.

**Comment by arantes555** [comment]
@vbourgeois I just released `electron-fetch@1.4.0`, which should fix the biggest issues with `electron@7`. However, a number of problems are still present. Off the top of my head: 
- impossible to read body of a response with an error status
- impossible to read cookies in response headers without some tricks
- stricter forbidden headers in requests (like impossible to set a `Host` header)
- some weird things with unusual redirects (this one is not very important I think)
- weird behaviour with proxies (this one needs further investigation)
- weird crashes at seemingly random moments, often after a failed request (obviously needs further investigation)

However, basic requests should now work, not just in `GET` anymore :)

**Comment by vbourgeois** [comment]
Great, thank you!
